AlloySmelter.addRecipe cause a crash the game with last version of ender IO (1.12.2-5.3.68)

After updating Ender IO in last version, this line in my script throw an exception on game launch (in phase 4 of 7 of forge loader) 

Script line causing this issue :
`mods.enderio.AlloySmelter.addRecipe(<minecraft:iron_nugget> * 9, [<exnihilocreatio:item_ore_iron:1>], 2000);`

This line work perfectly before updating Ender IO in version 1.12.2-5.3.68

Crash report : 
net.minecraftforge.fml.common.LoaderExceptionModCrash: Caught exception from EnderTweaker (endertweaker)
Caused by: java.lang.NoSuchMethodError: crazypants.enderio.base.recipe.alloysmelter.AlloyRecipeManager.addRecipe(Lcrazypants/enderio/base/recipe/Recipe;)V
    at shadows.endertweaker.AlloySmelter.lambda$addRecipe$0(AlloySmelter.java:37)
    at shadows.endertweaker.AlloySmelter$$Lambda$4146/2140640543.run(Unknown Source)
    at shadows.endertweaker.EnderTweaker.loadComplete(EnderTweaker.java:30)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:497)
    at net.minecraftforge.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:637)
    at sun.reflect.GeneratedMethodAccessor3.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:497)
    at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91)
    at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150)
    at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76)
    at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399)
    at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71)
    at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116)
    at com.google.common.eventbus.EventBus.post(EventBus.java:217)
    at net.minecraftforge.fml.common.LoadController.sendEventToModContainer(LoadController.java:219)
    at net.minecraftforge.fml.common.LoadController.propogateStateMessage(LoadController.java:197)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:497)
    at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91)
    at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150)
    at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76)
    at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399)
    at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71)
    at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116)
    at com.google.common.eventbus.EventBus.post(EventBus.java:217)
    at net.minecraftforge.fml.common.LoadController.distributeStateMessage(LoadController.java:136)
    at net.minecraftforge.fml.common.Loader.initializeMods(Loader.java:757)
    at net.minecraftforge.fml.client.FMLClientHandler.finishMinecraftLoading(FMLClientHandler.java:336)
    at net.minecraft.client.Minecraft.init(Minecraft.java:535)
    at net.minecraft.client.Minecraft.run(Minecraft.java:3931)
    at net.minecraft.client.main.Main.main(SourceFile:123)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:497)
    at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
    at net.minecraft.launchwrapper.Launch.main(Launch.java:28)

Thank's for your help.

--- a/build.gradle
+++ b/build.gradle
@@ -44,7 +44,7 @@ repositories {
 dependencies {
 	deobfCompile "mezz.jei:jei_1.12.2:+"
 	deobfCompile "mcp.mobius.waila:Hwyla:1.8.+"
-	deobfCompile("com.enderio:EnderIO:1.12.2-5.2.66"){
+	deobfCompile("com.enderio:EnderIO:1.12.2-5.3.68"){
         transitive = false
     }
 	deobfCompile "com.enderio.core:EnderCore:1.12.2-+"

--- a/src/main/java/shadows/endertweaker/AlloySmelter.java
+++ b/src/main/java/shadows/endertweaker/AlloySmelter.java
@@ -1,28 +1,23 @@
 package shadows.endertweaker;
 
-import java.util.ArrayList;
-import java.util.List;
-
 import com.enderio.core.common.util.NNList;
-
 import crafttweaker.CraftTweakerAPI;
 import crafttweaker.annotations.ZenRegister;
 import crafttweaker.api.item.IIngredient;
 import crafttweaker.api.item.IItemStack;
 import crafttweaker.api.minecraft.CraftTweakerMC;
 import crazypants.enderio.base.recipe.IManyToOneRecipe;
-import crazypants.enderio.base.recipe.IRecipe;
 import crazypants.enderio.base.recipe.MachineRecipeInput;
-import crazypants.enderio.base.recipe.RecipeBonusType;
 import crazypants.enderio.base.recipe.RecipeLevel;
-import crazypants.enderio.base.recipe.RecipeOutput;
 import crazypants.enderio.base.recipe.alloysmelter.AlloyRecipeManager;
 import net.minecraft.item.ItemStack;
 import net.minecraftforge.oredict.OreDictionary;
-import shadows.endertweaker.recipe.ManyToOneRecipe;
 import stanhebben.zenscript.annotations.Optional;
 import stanhebben.zenscript.annotations.ZenClass;
 import stanhebben.zenscript.annotations.ZenMethod;
+
+import java.util.ArrayList;
+import java.util.List;
 
 @ZenClass("mods.enderio.AlloySmelter")
 @ZenRegister
@@ -31,11 +26,8 @@ public class AlloySmelter {
 	@ZenMethod
 	public static void addRecipe(IItemStack output, IIngredient[] input, @Optional int energyCost, @Optional float xp) {
 		if (hasErrors(output, input)) return;
-		EnderTweaker.ADDITIONS.add(() -> {
-			RecipeOutput out = new RecipeOutput(CraftTweakerMC.getItemStack(output), 1, xp);
-			ManyToOneRecipe rec = new ManyToOneRecipe(out, energyCost, RecipeBonusType.NONE, RecipeLevel.IGNORE, RecipeUtils.toEIOInputs(input));
-			AlloyRecipeManager.getInstance().addRecipe(rec);
-		});
+		EnderTweaker.ADDITIONS.add(() -> AlloyRecipeManager.getInstance().addRecipe(false, new NNList<>(RecipeUtils.toEIOInputs(input)),
+				CraftTweakerMC.getItemStack(output), energyCost, xp, RecipeLevel.IGNORE));
 	}
 
 	@ZenMethod
@@ -69,7 +61,7 @@ public class AlloySmelter {
 			for (int i = 0; i < input.length; i++) {
 				inputs.add(new MachineRecipeInput(i, CraftTweakerMC.getItemStack(input[i])));
 			}
-			IRecipe r = AlloyRecipeManager.getInstance().getRecipeForInputs(RecipeLevel.IGNORE, inputs);
+			IManyToOneRecipe r = (IManyToOneRecipe)AlloyRecipeManager.getInstance().getRecipeForInputs(RecipeLevel.IGNORE, inputs);
 
 			if (r != null) {
 				AlloyRecipeManager.getInstance().getRecipes().remove(r);

--- a/src/main/java/shadows/endertweaker/Enchanter.java
+++ b/src/main/java/shadows/endertweaker/Enchanter.java
@@ -52,7 +52,8 @@ public class Enchanter {
 		EnderTweaker.REMOVALS.add(() -> {
 			Enchantment ench = (Enchantment) output.getInternal();
 			String id = null;
-			for (Map.Entry<String, IMachineRecipe> ent : MachineRecipeRegistry.instance.getRecipesForMachine(MachineRecipeRegistry.ENCHANTER).entrySet()) {
+			for (Map.Entry<String, ? extends IMachineRecipe> ent :
+			MachineRecipeRegistry.instance.getRecipesForMachine(MachineRecipeRegistry.ENCHANTER).entrySet()) {
 				if (((EnchanterRecipe) ent.getValue()).getEnchantment() == ench) {
 					id = ent.getKey();
 					break;

--- a/src/main/java/shadows/endertweaker/SoulBinder.java
+++ b/src/main/java/shadows/endertweaker/SoulBinder.java
@@ -25,9 +25,9 @@ public class SoulBinder {
 
 	@ZenMethod
 	public static void addRecipe(IItemStack output, IIngredient input, String[] entities, int xp, @Optional int energyCost) {
-		EnderTweaker.ADDITIONS.add(() -> {
-			MachineRecipeRegistry.instance.registerRecipe(new SoulBinderRecipe(CraftTweakerMC.getIngredient(input), CraftTweakerMC.getItemStack(output), xp, energyCost <= 0 ? 5000 : energyCost, RecipeLevel.IGNORE, toRelocs(entities)));
-		});
+		EnderTweaker.ADDITIONS.add(() -> MachineRecipeRegistry.instance.registerRecipe(
+				new SoulBinderRecipe(CraftTweakerMC.getIngredient(input), CraftTweakerMC.getItemStack(output), xp,
+						energyCost <= 0 ? 5000 : energyCost, RecipeLevel.IGNORE, toRelocs(entities))));
 	}
 
 	@ZenMethod
@@ -39,7 +39,7 @@ public class SoulBinder {
 		EnderTweaker.REMOVALS.add(() -> {
 			ItemStack stack = CraftTweakerMC.getItemStack(output);
 			String id = null;
-			for (Map.Entry<String, IMachineRecipe> ent : MachineRecipeRegistry.instance.getRecipesForMachine(MachineRecipeRegistry.SOULBINDER).entrySet()) {
+			for (Map.Entry<String, ? extends IMachineRecipe> ent : MachineRecipeRegistry.instance.getRecipesForMachine(MachineRecipeRegistry.SOULBINDER).entrySet()) {
 				if (OreDictionary.itemMatches(stack, ((ISoulBinderRecipe) ent.getValue()).getOutputStack(), false)) {
 					id = ent.getKey();
 					break;

--- a/src/main/java/shadows/endertweaker/Tank.java
+++ b/src/main/java/shadows/endertweaker/Tank.java
@@ -45,7 +45,8 @@ public class Tank {
 		EnderTweaker.REMOVALS.add(() -> {
 			String rec = null;
 			String id = fill ? MachineRecipeRegistry.TANK_FILLING : MachineRecipeRegistry.TANK_EMPTYING;
-			for (Map.Entry<String, IMachineRecipe> ent : MachineRecipeRegistry.instance.getRecipesForMachine(id).entrySet()) {
+			for (Map.Entry<String, ? extends IMachineRecipe> ent :
+			MachineRecipeRegistry.instance.getRecipesForMachine(id).entrySet()) {
 				TankMachineRecipe r = (TankMachineRecipe) ent.getValue();
 				if (r.getFluid().getFluid().getName().equals(fluid.getName()) && OreDictionary.itemMatches(CraftTweakerMC.getItemStack(output), r.getOutput().getMatchingStacks()[0], false)) {
 					rec = ent.getKey();

